src/tomczyk-kamil/Fsd.Kamil.Client/Fsd.Kamil.Ng.Ex1/restaurant-app/src/app/restaurant/restaurant-products/restaurant-products.component.ts

```diff
@@ -0,0 +1,65 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { Product } from '../models/product';
+import { ChoiceService } from '../choice/choice.service';
+import { ProductChoice } from '../models/product-choice';
+
+@Component({
+  selector: 'app-restaurant-products',
+  templateUrl: './restaurant-products.component.html',
+  styleUrls: ['./restaurant-products.component.css']
+})
+export class RestaurantProductsComponent implements OnInit {
+  buttonClass: string;
+  isTableShown: boolean;
+  statusText: string;
+  choiceService: ChoiceService;
+  @Input() drinks: Product[];
+  @Input() food: Product[];
+  @Input() choice: ProductChoice[];
+
+  constructor(choiceService: ChoiceService) {
+    this.choiceService = choiceService;
+    this.choice = [];
+  }
+
+  ngOnInit(): void {
+    this.buttonClass = 'btn';
+    this.changeTableVisibility();
+
+    this.food = [
+      new Product(1, "Pizza", "Ser, szynka, pieczarki", 25, 1),
+      new Product(2, "Pierogi", "Z owocami", 14, 1),
+      new Product(3, "Bigos", "Kapusta, kiełbasa", 12, 1),
+      new Product(4, "Hamburger", "Mięso, cebula, sałata, sos", 22, 1)
+    ];
+
+    this.drinks = [
+      new Product(5, "Woda mineralna", "Jurajska", 3, 1),
+      new Product(6, "Sok", "Pomarańczowy", 4, 1),
+      new Product(7, "Wino", "Dobre", 12, 1)
+    ];
+  }
+
+  changeTableVisibility() {
+    this.isTableShown = !this.isTableShown;
+    this.statusText = this.isTableShown ? 'Menu' : 'Ukryj'; //nie było w założeniach, w ramach ćwiczenia z zajęć
+  }
+
+  addActionHasBeenDone(order: Product) {
+    let newOrderedProduct: ProductChoice = new ProductChoice(order.name, order.price, 1);
+    let existingProduct = this.choiceService.choices.find(o => o.name === newOrderedProduct.name) // odnalezienie produktu,do którego wpisujemy nową ilość
+    //Najgorsza część - znalezienie tego produtku, żeby tam dodać ilość
```

Comment:
zabrakło Ci dobrej architektury domeny biznesowej tzn. rozrysowania na kartce jak będzie wyglądał produkt itp. ja sugeruje rozwiązanie z klasami: Produkt, Danie, Napój, Zamówienie. Danie i Napój dziedziczą z Produkt. Zamówienie ma w sobie 2 pola: Produkt i ilość sztuk, a także 1 metodę policzWartośćZamówienia(). W jednym serwisie powinieneś trzymać listę zamówień. Nie widzę u Ciebie encji "Zamówienie"

Reply:
Dzięki za uwagi. Postarałem się to zmienić zgodnie z sugestią - rzeczywiście takie narysowanie pomaga.